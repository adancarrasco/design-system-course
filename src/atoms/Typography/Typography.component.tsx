import * as React from 'react'
import styled from '@emotion/styled'

export type TypographyType = 'uppercase' | 'strong' | 'default'

export interface TypographyCSSProps {
  color?: string
  type?: TypographyType
}

export interface TypographyComponentProps extends TypographyCSSProps {
  children: React.ReactNode | string
}

const Typography: React.FC<TypographyComponentProps> = ({
  children,
  type = 'default',
  color = '#3d3d3d',
}) => {
  const TypographyComponent = typographyTypes[type]
  return <TypographyComponent color={color}>{children}</TypographyComponent>
}

const AllCaps = styled.span<TypographyCSSProps>`
  text-transform: uppercase;
  color: ${(props) => props.color};
`

const Strong = styled.span<TypographyCSSProps>`
  color: ${(props) => props.color};
  font-weight: bold;
`

const Default = styled.span<TypographyCSSProps>`
  color: ${(props) => props.color};
`
const typographyTypes = {
  uppercase: AllCaps,
  strong: Strong,
  default: Default,
}

export default Typography
