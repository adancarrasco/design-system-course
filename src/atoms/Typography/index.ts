export { default } from './Typography.component'
export type { TypographyComponentProps, TypographyType } from './Typography.component'
