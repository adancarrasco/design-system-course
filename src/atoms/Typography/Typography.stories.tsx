import React from 'react'
import { storiesOf } from '@storybook/react'

import Typography from '.'

storiesOf('Atoms|Typography', module)
  .add('Default', () => <Typography>Test</Typography>)
  .add('Uppercase', () => <Typography type="uppercase">All caps here</Typography>)
  .add('Strong', () => <Typography type="strong">Strong text here</Typography>)
