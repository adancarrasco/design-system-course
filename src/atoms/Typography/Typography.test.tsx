import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import Typography, { TypographyType } from '.'

describe('Typography Component', () => {
  const renderComponent = (type?: TypographyType) =>
    render(<Typography type={type}>Hello</Typography>)

  it('matches snapshot', () => {
    const { getByText } = renderComponent()
    const container = getByText('Hello')
    expect(container).toMatchSnapshot()
  })

  it('should render', () => {
    const { getByText } = renderComponent()
    const container = getByText('Hello')
    expect(container).toBeInTheDocument()
  })

  it('should render as all caps', () => {
    const { getByText } = renderComponent('strong')
    const container = getByText('Hello')
    expect(container).toHaveStyleRule('font-weight', 'bold')
  })

  it('should render as uppercase', () => {
    const { getByText } = renderComponent('uppercase')
    const container = getByText('Hello')
    expect(container).toHaveStyleRule('text-transform', 'uppercase')
  })
})
