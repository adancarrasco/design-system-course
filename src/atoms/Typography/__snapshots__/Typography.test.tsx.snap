// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Component matches snapshot 1`] = `
<span
  class="css-1edki26"
>
  Hello
</span>
`;
